File: Programmers/LEVEL_1/제일 작은 수 제거하기.cpp
#include <string>
#include <vector>
#include <limits.h>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    int size = arr.size();
    int check = INT_MAX;
    int index;
    for (int i = 0; i < size; i++)
    {
        if (arr[i] < check)
        {
            check = arr[i];
        }
    }

    for (int i = 0; i < size; i++)
    {
        if (arr[i] == check)
            continue;
        else
            answer.push_back(arr[i]);
    }

    if (answer.empty())
    {
        answer.push_back(-1);
    }
    return answer;
}

File: Programmers/LEVEL_1/제일 작은 수 제거하기.js
function solution(arr) {
  var answer = [];

  if (arr.length == 1) {
    answer.push(-1);
  }

  let min = Math.min.apply(null, arr);

  for (let i of arr) {
    if (i == min) continue;
    answer.push(i);
  }
  // splice, indexOf 쓰면 더 간단할듯
  return answer;
}
